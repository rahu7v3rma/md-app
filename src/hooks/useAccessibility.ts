import { useEffect, useState } from 'react';
import { Dimensions } from 'react-native';
import { getFontScale } from 'react-native-device-info';

const { height: windowHeight, width: windowWidth } = Dimensions.get('window');

const useAccessibility = () => {
    // note: font scale will currently only be set once per hook-usage
    const [fontScale, setFontScale] = useState<number | null>(null);

    useEffect(() => {
        getFontScale().then((scale) => setFontScale(scale));
    }, []);

    return {
        fontScale: Number(fontScale),
        windowHeight,
        windowWidth
    };
};

export default useAccessibility;
