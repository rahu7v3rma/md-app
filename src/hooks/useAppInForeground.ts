import { useEffect, useRef } from 'react';
import { AppState } from 'react-native';

/* this hooks will run a callback function every time the app returns to the
 * forground, either from the background or from being inactive
 */
const useAppInForeground = (callback: () => void) => {
    const appState = useRef(AppState.currentState);

    // when the hook is just called we should be in the foreground
    callback();

    useEffect(() => {
        const subscription = AppState.addEventListener(
            'change',
            (nextAppState) => {
                if (
                    appState.current.match(/inactive|background/) &&
                    nextAppState === 'active'
                ) {
                    callback();
                }

                appState.current = nextAppState;
            }
        );

        return () => {
            subscription.remove();
        };
    }, [callback]);
};

export default useAppInForeground;
