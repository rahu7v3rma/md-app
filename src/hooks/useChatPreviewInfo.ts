import { useEffect, useState } from 'react';
import type { Channel } from 'stream-chat';
import { useChatContext } from 'stream-chat-react-native';

import {
    getChannelDisplayImage,
    getChannelDisplayTitle,
    getChannelIsOnline,
    getChannelMembers
} from '@/services/chat';

type Props = {
    channel: Channel;
};

const useChannelPreviewInfo = ({ channel }: Props) => {
    const { client } = useChatContext();

    const [displayTitle, setDisplayTitle] = useState<string | undefined>(
        getChannelDisplayTitle(channel, client.user)
    );

    const [displayImage, setDisplayImage] = useState<string | undefined>(
        getChannelDisplayImage(channel, client.user)
    );

    const [isOnline, setIsOnline] = useState<boolean>(
        getChannelIsOnline(channel, client.user)
    );

    const [members, setMembers] = useState<number>(getChannelMembers(channel));

    useEffect(() => {
        const handleEvent = () => {
            setDisplayTitle((currentDisplayTitle) => {
                const newDisplayTitle = getChannelDisplayTitle(
                    channel,
                    client.user
                );
                return currentDisplayTitle !== newDisplayTitle
                    ? newDisplayTitle
                    : currentDisplayTitle;
            });
            setDisplayImage((currentDisplayImage) => {
                const newDisplayImage = getChannelDisplayImage(
                    channel,
                    client.user
                );
                return currentDisplayImage !== newDisplayImage
                    ? newDisplayImage
                    : currentDisplayImage;
            });
            setIsOnline(getChannelIsOnline(channel, client.user));
            setMembers(getChannelMembers(channel));
        };

        client.on('channel.updated', handleEvent);
        channel.watch();

        return () => {
            channel.stopWatching();
            client.off('channel.updated', handleEvent);
        };
    }, [client, channel]);

    return {
        displayImage: displayImage,
        displayTitle: displayTitle,
        isOnline: isOnline,
        totalMembers: members
    };
};

export default useChannelPreviewInfo;
