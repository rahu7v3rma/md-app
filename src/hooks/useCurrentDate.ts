import moment from 'moment';
import { useCallback, useEffect, useState } from 'react';

const useCurrentDate = () => {
    const [currentDate, setCurrentDate] = useState<Date>();
    const [currentDateFormatted, setCurrentDateFormatted] = useState<string>();

    const updateCurrentDate = useCallback(() => {
        const updatedDate = new Date();
        setCurrentDate(updatedDate);
        setCurrentDateFormatted(moment(new Date()).format('YYYY-MM-DD'));
    }, []);

    const scheduleCurrentDateUpdate = useCallback(() => {
        const d = new Date();
        const secondsToEndOfDay =
            24 * 60 * 60 -
            d.getHours() * 60 * 60 -
            d.getMinutes() * 60 -
            d.getSeconds();

        // schedule a date update one second after the end of this day
        return setTimeout(() => {
            updateCurrentDate();
        }, (secondsToEndOfDay + 1) * 1000);
    }, [updateCurrentDate]);

    useEffect(() => {
        updateCurrentDate();

        let timeoutRef = scheduleCurrentDateUpdate();

        // reschedule the timeout every hour so even if the app is running
        // as the date changes we will update the next time the date changes
        // as well
        const rescheduleInterval = setInterval(() => {
            if (timeoutRef) {
                clearTimeout(timeoutRef);
            }

            timeoutRef = scheduleCurrentDateUpdate();
        }, 60 * 60 * 1000);

        return () => {
            clearInterval(rescheduleInterval);

            if (timeoutRef) {
                clearTimeout(timeoutRef);
            }
        };
    }, [updateCurrentDate, scheduleCurrentDateUpdate]);

    return { currentDate, currentDateFormatted };
};

export default useCurrentDate;
