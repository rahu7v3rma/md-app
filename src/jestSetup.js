import mockRNCameraRoll from '@react-native-camera-roll/camera-roll/src/__mocks__/nativeInterface';
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock';
import mockRNDeviceInfo from 'react-native-device-info/jest/react-native-device-info-mock';

import { MockData } from './utils/mockData';

// setImmediate and clearImmediate aren't mocked by useFakeTimers if they are
// not set to a function (https://github.com/jestjs/jest/pull/11599)
global.setImmediate = jest.fn();
global.clearImmediate = jest.fn();
jest.useRealTimers();

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

jest.mock('react-native-keychain', () => {
    return {
        SECURITY_LEVEL_ANY: 'MOCK_SECURITY_LEVEL_ANY',
        SECURITY_LEVEL_SECURE_SOFTWARE: 'MOCK_SECURITY_LEVEL_SECURE_SOFTWARE',
        SECURITY_LEVEL_SECURE_HARDWARE: 'MOCK_SECURITY_LEVEL_SECURE_HARDWARE',
        setGenericPassword: jest.fn().mockResolvedValue(),
        getGenericPassword: jest.fn().mockResolvedValue(false),
        resetGenericPassword: jest.fn().mockResolvedValue()
    };
});

jest.mock('react-native-splash-screen', () => {
    return {
        hide: jest.fn()
    };
});

export let mockUnwrap = jest.fn().mockImplementation(() => {
    return Promise.resolve({});
});
export const mockDispatch = jest.fn().mockImplementation(() => {
    return {
        unwrap: mockUnwrap
    };
});
jest.mock('@/hooks', () => ({
    useAppDispatch: () => mockDispatch,
    useChannelPreviewInfo: () => mockDispatch
}));
export const mockChannelId = 'mockChannelId';
export const mockChatClient = jest.fn().mockImplementation(() => ({
    userID: 1,
    chat_id: 1,
    queryChannels: jest.fn().mockImplementation(() => {
        return Promise.resolve([
            {
                cid: 99
            }
        ]);
    }),
    activeChannels: {
        mockChannelId
    }
}));
jest.mock('@/contexts/appChat', () => ({
    useAppChat: () => {
        return {
            chatClient: mockChatClient(),
            setActiveChatChannel: jest.fn()
        };
    }
}));
export const mockRNSha256 = jest
    .fn()
    .mockImplementation(() => Promise.resolve('123'));
jest.mock('react-native-sha256', () => {
    return {
        sha256: mockRNSha256
    };
});

jest.mock('react-native-share', () => ({
    default: jest.fn()
}));

jest.mock('@/services/api', () => ({
    getImageUploadCredentials: jest.fn(() => ({
        keyPrefix: 'keyPrefix',
        fileName: 'fileName',
        bucketName: 'bucketName'
    }))
}));

jest.mock('@/reducers/content', () => ({
    ContentSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        journey: MockData.journey,
        navigationKey: 'Main.Home',
        activeBlock: MockData.activeBlock,
        activeLesson: {
            lesson: {}
        },
        journeyBlocks: MockData.journeyBlocks
    })),
    loadJourney: jest.fn(),
    getActiveLesson: jest.fn()
}));

export const mockChatProfile = jest.fn(() => ({
    apiKey: 'apiKey',
    token: 'token',
    userId: 'userId',
    firebasePushProviderName: 'firebasePushProviderName'
}));
export const mockDeleteAccount = jest.fn();
jest.mock('@/reducers/user', () => ({
    UserSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        userInfo: {
            username: 'user1',
            first_name: 'Some',
            last_name: 'Name',
            email: 'user1@example.example'
        },
        userProfile: {
            diabetes_type: 'Prediabetes',
            image: null,
            onboarding_form_url: null
        },
        coach: {
            first_name: 'name',
            profile_image: 'profile_image'
        },
        chatProfile: mockChatProfile(),
        hasCoachChat: false
    })),
    userLogin: jest.fn(),
    changePassword: jest.fn(),
    resetPassword: jest.fn(),
    resetPasswordConfirm: jest.fn(),
    resetPasswordVerify: jest.fn(),
    getCoach: jest.fn(),
    refreshProfileSession: jest.fn(),
    getProfile: jest.fn(),
    searchFilterAction: jest.fn(),
    clearSearchText: jest.fn(),
    deleteAccount: mockDeleteAccount
}));
export const mockRNFSExists = jest
    .fn()
    .mockImplementation(() => Promise.resolve(false));
export const mockRNFSCachesDirectoryPath = 'mockCachesDirectoryPath';
export const mockRNFSDownloadFile = jest.fn().mockImplementation(() => ({
    promise: Promise.resolve({ statusCode: 200 })
}));
jest.mock('react-native-fs', () => {
    return {
        readFile: jest.fn(),
        exists: mockRNFSExists,
        CachesDirectoryPath: mockRNFSCachesDirectoryPath,
        downloadFile: mockRNFSDownloadFile,
        unlink: jest.fn().mockImplementation(() => Promise.resolve())
    };
});
jest.mock('react-native-image-crop-picker', () => {
    return {
        openPicker: jest.fn(),
        launchImageLibrary: jest.fn(),
        openCamera: jest.fn()
    };
});

jest.mock('@react-native-camera-roll/camera-roll', () => mockRNCameraRoll);
jest.mock('react-native-reanimated', () => ({
    ...require('react-native-reanimated/mock')
}));
jest.mock('react-native-device-info', () => mockRNDeviceInfo);
jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');

jest.mock('@/reducers/track', () => ({
    TrackSelectors: jest.fn().mockImplementation(() => ({
        trackFastState: '',
        trackStartedAt: '',
        trackTimeLimitInSeconds: 57600,
        trackedTimeInSeconds: 0
    })),
    trackStart: jest.fn(),
    trackUpdate: jest.fn()
}));

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo);

jest.mock('@notifee/react-native', () =>
    require('@notifee/react-native/jest-mock')
);

export const mockOnTokenRefresh = jest.fn();
jest.mock('@react-native-firebase/messaging', () => ({
    __esModule: true,
    default: () => ({ onTokenRefresh: mockOnTokenRefresh }),
    FirebaseMessagingTypes: jest.fn()
}));

export const mockNavigate = jest.fn();
export const mockPop = jest.fn();
export const mockGoBack = jest.fn();
export const mockReplace = jest.fn();
export const mockNavigationDispatch = jest.fn();
export const mockUseRoute = jest.fn().mockImplementation(() => ({
    params: {
        id: 1,
        unit: 'kg',
        code: '11111',
        url: 'https://www.google.com/'
    }
}));

jest.mock('@/services/notification', () => ({
    resetNotifications: jest.fn(),
    registerBackgroundNotificationsListener: jest.fn(),
    getFCMToken: jest.fn().mockImplementation(() => {
        return Promise.resolve({});
    }),
    triggerFirebaseForegroundNotification: jest.fn(),
    registerFirebaseMessageListener: jest.fn(),
    requestNotificationPermissions: jest.fn()
}));

jest.mock('@react-navigation/native', () => {
    const real = jest.requireActual('@react-navigation/native');
    return {
        ...real,
        useNavigation: jest.fn().mockImplementation(() => {
            return {
                canGoBack: jest.fn().mockImplementation(() => {
                    return true;
                }),
                goBack: mockGoBack,
                navigate: mockNavigate,
                pop: mockPop,
                replace: mockReplace,
                dispatch: mockNavigationDispatch
            };
        }),
        useRoute: mockUseRoute,
        useIsFocused: jest.fn().mockImplementation(() => ({
            isFocused: true
        })),
        useFocusEffect: jest.fn()
    };
});

export const mockToast = jest.fn();

jest.mock('react-native-toast-message', () => ({
    show: mockToast
}));

export const mockGetChannelDisplayImage = jest.fn();
jest.mock('@/services/chat', () => ({
    getChannelDisplayImage: mockGetChannelDisplayImage
}));

export const mockConnectUser = jest.fn(() => Promise.resolve());
export const mockDisconnectUser = jest.fn();
export const mockGetInstance = jest.fn(() => ({
    connectUser: mockConnectUser,
    disconnectUser: mockDisconnectUser
}));
jest.mock('stream-chat', () => ({
    StreamChat: {
        getInstance: mockGetInstance
    }
}));

jest.mock('@aws-sdk/client-s3', () => ({
    S3Client: jest.fn(() => ({
        send: jest.fn(() => Promise.resolve())
    })),
    PutObjectCommand: jest.fn().mockImplementation(() => {
        return {};
    })
}));

jest.mock('react-native-pdf', () => null);
