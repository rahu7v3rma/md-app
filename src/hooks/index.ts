import useAppDispatch from './useAppDispatch';
import useAppInForeground from './useAppInForeground';
import useChannelPreviewInfo from './useChatPreviewInfo';
import useCurrentDate from './useCurrentDate';

export {
    useAppDispatch,
    useAppInForeground,
    useChannelPreviewInfo,
    useCurrentDate
};
