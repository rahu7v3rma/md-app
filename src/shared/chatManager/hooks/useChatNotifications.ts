import messaging, {
    FirebaseMessagingTypes
} from '@react-native-firebase/messaging';
import { useCallback, useEffect, useRef } from 'react';
import { StreamChat } from 'stream-chat';

import { useAppDispatch } from '@/hooks';
import { refreshProfileSession, UserSelectors } from '@/reducers/user';
import {
    getFCMToken,
    registerFirebaseMessageListener,
    triggerFirebaseForegroundNotification
} from '@/services/notification';
import { getCachedFCMPushToken, setCachedFCMPushToken } from '@/utils/storage';

const registerChatPushToken = async (
    chatClient: StreamChat,
    firebasePushProviderName: string,
    onNewToken?: (token: string) => void
) => {
    const token = await getFCMToken();
    const pushProvider = 'firebase';

    // add and cache current token
    await chatClient.addDevice(
        token,
        pushProvider,
        chatClient.userID,
        firebasePushProviderName
    );
    await setCachedFCMPushToken(token);

    // the return value is an unsubscribe function for the event
    const unsubscribe = messaging().onTokenRefresh(async (newToken: string) => {
        // remove old token
        const oldToken = await getCachedFCMPushToken();
        if (oldToken !== null) {
            await chatClient.removeDevice(oldToken);
        }

        // add and cache new token
        await chatClient.addDevice(
            newToken,
            pushProvider,
            chatClient.userID,
            firebasePushProviderName
        );
        await setCachedFCMPushToken(newToken);

        // notify new token
        onNewToken?.(newToken);
    });

    return unsubscribe;
};

const useChatNotifications = (
    chatClient: StreamChat | null,
    activeChatChannel: string | null
) => {
    const dispatch = useAppDispatch();

    const { chatProfile } = UserSelectors();

    const unsubscribeTokenRefreshListenerRef = useRef<() => void>();
    const unsubscribeMessageListener = useRef<() => void>();

    const handleForegroundMessageReceived = useCallback(
        (remoteMessage: FirebaseMessagingTypes.RemoteMessage) => {
            if (remoteMessage.data?.cid !== activeChatChannel) {
                triggerFirebaseForegroundNotification(remoteMessage);
            }
        },
        [activeChatChannel]
    );

    useEffect(() => {
        const onNewToken = (token: string) => {
            dispatch(refreshProfileSession({ fcmToken: token }));
        };

        if (chatClient && chatProfile) {
            registerChatPushToken(
                chatClient,
                chatProfile.firebasePushProviderName,
                onNewToken
            )
                .then((unsubsribe) => {
                    unsubscribeTokenRefreshListenerRef.current = unsubsribe;
                })
                .catch((error) => {
                    console.log(
                        `An error occurred while registering Firebase push token: ${error?.message}`
                    );
                });
        }

        return () => {
            // unsubscribe token refresh listeners
            unsubscribeTokenRefreshListenerRef.current?.();
        };
    }, [dispatch, chatClient, chatProfile]);

    useEffect(() => {
        unsubscribeMessageListener.current = registerFirebaseMessageListener(
            handleForegroundMessageReceived
        );

        return () => {
            // unsubscribe message listeners
            unsubscribeMessageListener.current?.();
        };
    }, [handleForegroundMessageReceived]);

    return {
        chatClient
    };
};

export default useChatNotifications;
