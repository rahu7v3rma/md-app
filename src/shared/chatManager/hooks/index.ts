import useChatNotifications from './useChatNotifications';
import useInitChatClient from './useInitChatClient';

export { useChatNotifications, useInitChatClient };
