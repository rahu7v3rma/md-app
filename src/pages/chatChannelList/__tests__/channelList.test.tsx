import moment from 'moment';
import React, { ElementType } from 'react';
import renderer, { act } from 'react-test-renderer';
import { Channel } from 'stream-chat';
import {
    DefaultStreamChatGenerics,
    LatestMessagePreview
} from 'stream-chat-react-native';

import * as Hooks from '@/hooks';
import { mockNavigate } from '@/jestSetup';

import ChannelList from '../components/channelList';

jest.mock('stream-chat-react-native');

jest.unmock('@/hooks');

const mockLatestMessagePreview = {
    messageObject: {
        created_at: '2023-10-26T11:49:25.552Z'
    }
};

const mockChannel = {
    cid: 'message:1000'
};

jest.mock('@/hooks', () => ({
    useChannelPreviewInfo: jest.fn().mockReturnValue({
        displayImage: undefined,
        displayTitle: 'Coach',
        isOnline: true
    })
}));

describe('channelList tests', () => {
    let tree: renderer.ReactTestRenderer;

    it('Match Snapshot', () => {
        tree = renderer.create(
            <ChannelList
                channel={
                    mockChannel as any as Channel<DefaultStreamChatGenerics>
                }
                latestMessagePreview={
                    mockLatestMessagePreview as any as LatestMessagePreview<DefaultStreamChatGenerics>
                }
                unread={10}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    it('renders channel information correctly', () => {
        tree = renderer.create(
            <ChannelList
                channel={jest.fn() as any as Channel<DefaultStreamChatGenerics>}
                latestMessagePreview={
                    mockLatestMessagePreview as any as LatestMessagePreview<DefaultStreamChatGenerics>
                }
                unread={10}
            />
        );
        expect(tree.root.findByProps({ testID: 'online' })).toBeTruthy();
        expect(
            tree.root.findByProps({ testID: 'displayTitle' }).props.children
        ).toEqual('Coach');
        expect(
            tree.root.findByType('Image' as ElementType).props.source.testUri
        ).toEqual('../../../src/assets/images/default_avatar.jpg');
    });

    it('when user is offline', () => {
        jest.spyOn(Hooks, 'useChannelPreviewInfo').mockImplementation(() => ({
            displayImage: undefined,
            displayTitle: 'Coach',
            isOnline: false,
            totalMembers: 14
        }));
        tree = renderer.create(
            <ChannelList
                channel={
                    mockChannel as any as Channel<DefaultStreamChatGenerics>
                }
                latestMessagePreview={
                    mockLatestMessagePreview as any as LatestMessagePreview<DefaultStreamChatGenerics>
                }
                unread={10}
            />
        );
        expect(tree.root.findAllByProps({ testID: 'online' }).length).toEqual(
            0
        );
    });

    it('navigates to the ChatChannel screen on press', async () => {
        tree = renderer.create(
            <ChannelList
                channel={
                    mockChannel as any as Channel<DefaultStreamChatGenerics>
                }
                latestMessagePreview={
                    mockLatestMessagePreview as any as LatestMessagePreview<DefaultStreamChatGenerics>
                }
                unread={10}
            />
        );
        await act(() => (tree.root.children[0] as any).props.onPress());
        expect(mockNavigate).toBeCalledWith('ChatChannel', {
            channelId: mockChannel.cid
        });
    });

    it('displays the correct latest message date format', async () => {
        tree = renderer.create(
            <ChannelList
                channel={
                    mockChannel as any as Channel<DefaultStreamChatGenerics>
                }
                latestMessagePreview={
                    mockLatestMessagePreview as any as LatestMessagePreview<DefaultStreamChatGenerics>
                }
                unread={10}
            />
        );

        expect(
            tree.root.findByProps({ testID: 'messageDate' }).props.children
        ).toEqual(
            moment(mockLatestMessagePreview.messageObject.created_at).format(
                'hh:mm A'
            )
        );
    });

    it('displays the unread count correctly', async () => {
        tree = renderer.create(
            <ChannelList
                channel={
                    mockChannel as any as Channel<DefaultStreamChatGenerics>
                }
                latestMessagePreview={
                    mockLatestMessagePreview as any as LatestMessagePreview<DefaultStreamChatGenerics>
                }
                unread={10}
            />
        );
        expect(
            tree.root.findByProps({ testID: 'unread' }).props.children
        ).toEqual(10);
    });
});
