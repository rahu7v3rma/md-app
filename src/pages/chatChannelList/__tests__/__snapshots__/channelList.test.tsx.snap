// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channelList tests Match Snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#cccccc",
        "borderRadius": 20,
        "elevation": 1,
        "flexDirection": "row",
        "height": 90,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 12,
        "paddingHorizontal": 20,
        "paddingVertical": 23,
        "shadowColor": "rgba(0,0,0,0.5)",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.06,
        "shadowRadius": 3,
      }
    }
  >
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/images/default_avatar.jpg",
          }
        }
        style={
          {
            "borderRadius": 16,
            "height": 32,
            "width": 32,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#4DBA0B",
            "borderColor": "#FFFFFF",
            "borderRadius": 12,
            "borderStyle": "solid",
            "borderWidth": 2,
            "height": 12,
            "position": "absolute",
            "top": 0,
            "transform": [
              {
                "translateY": -3,
              },
              {
                "translateX": -3,
              },
            ],
            "width": 12,
            "zIndex": 1,
          }
        }
        testID="online"
      />
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flex": 2.3,
          "marginLeft": 12,
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            undefined,
            {
              "color": "#271A51",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
          ]
        }
        testID="displayTitle"
      >
        Coach
      </Text>
      <View
        style={
          {
            "marginTop": 5,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
          "flex": 0.7,
        }
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        style={
          [
            {
              "fontFamily": "Poppins",
            },
            undefined,
            {
              "color": "#A4AAAF",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 20,
            },
          ]
        }
        testID="messageDate"
      >
        02:49 PM
      </Text>
      <View
        style={
          {
            "height": 16,
            "minWidth": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#72996B",
              "borderRadius": 8,
              "fontSize": 11,
              "height": 16,
              "justifyContent": "center",
              "marginTop": 3,
              "minWidth": 16,
              "padding": 3,
            }
          }
        >
          <Text
            adjustsFontSizeToFit={true}
            style={
              [
                {
                  "fontFamily": "Poppins",
                },
                undefined,
                {
                  "color": "#FFFFFF",
                  "fontSize": undefined,
                  "fontWeight": "500",
                  "lineHeight": undefined,
                },
              ]
            }
            testID="unread"
          >
            10
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
